src/wallet/rpcwallet.cpp

```diff
@@ -2318,12 +2318,17 @@ static UniValue settxfee(const JSONRPCRequest& request)
     }
 
             RPCHelpMan{"settxfee",
-                "\nSet the transaction fee per kB for this wallet. Overrides the global -paytxfee command line parameter.\n",
+                "\nSet the transaction fee per kB for this wallet. Overrides the global -paytxfee command line parameter.\n"
+                "Can be deactivated by passing 0 as the fee. In that case automatic fee selection will be used by default.\n",
```

Comment:
Nice documentation. Not sure on changing the format without an -rpcdeprecated cycle or whether it should be changed at all.

Reply:
Thanks, I agree the documentation is the minimum and it might be good enough. If the change in return the value is too much hassle or if there is not enough interest, I can remove it.

Reply:
I can't really see the value. Why would an interface return the value you passed in. Should the caller compare the result["feerate"] with the one passed in? Same goes for "active", which is simply the value passed in and cast to a bool.

Reply:
I am repeating a pattern I am seeing in `setwalletflag` and a few other RPCs. I see the value in being explicit what is actually going on, i.e. that setting to 0 means the parameter is not active. Do you have an alternative change you would prefer, for example returning the current fee rate or just the `active` status? Or would you prefer to not make any change to the return value?

Reply:
The current `return true` unconditionally is completely useless in the same sense. An API client wouldn't know what to do with the value. Should it check if it is false and then throw an error? In that case we might as well throw the error on the server.

So I can see your desire to clean it up, but the cost of breaking every API client for this doesn't seem justified. On top, just because we use that pattern in other calls, doesn't mean that it is a good pattern. I am not really aware of any setter method that returns the set value again, e.g. https://en.cppreference.com/w/cpp/atomic/atomic/store returns void.

> I see the value in being explicit what is actually going on

This is what documentation is for, because the RPC interface is mostly used by machines and not humans interpreting the response to get feedback on what was going.

Maybe what could make sense here is to return the fee reason in send* and fund* rpcs. Something like
-> sendtoaddress
<- { "txid" : "deadbeef", "fee_reason" : "paytxfee" }

Reply:
> Maybe what could make sense here is to return the fee reason in send* and fund* rpcs. Something like
> -> sendtoaddress
> <- { "txid" : "deadbeef", "fee_reason" : "paytxfee" }

Nice -- I think this could be a useful change.